src/AbstractOperations/conditional_operations.jl

```diff
@@ -86,11 +89,22 @@ function ConditionalOperation(operand::AbstractField;
     return ConditionalOperation{LX, LY, LZ}(operand, func, operand.grid, condition, mask)
 end
 
+validate_condition(cond::Function, ::ConditionalOperation) = cond
+
+function validate_condition(cond::AbstractArray, c::ConditionalOperation)
+    if ndims(cond) != 3
+        throw(ArgumentError("when kwarg condition::Array requires a 3D array of size $(size(c))"))
```

Comment:
i dont totally understand this, what do you mean "when" kwarg. Why just say "The keyword argument condition::AbstractArray requires..."

Note this does not require an array specifically. It can be a `Field`, or an `AbstractOperation` (or a `CuArray`)

Reply:
the dispatch is specifically on `AbstractArray`, so the error is correct to point out exactly that

Reply:
hm... should I just leave the `cond` untyped?

Reply:
Oh I got what you mean! you are referring to the error being Array-specific!